calibration_assistant.cpp

```diff
@@ -0,0 +1,1266 @@
+}
+
+void CalibrationAssistant::OnCustom(wxCommandEvent& evt)
+{
+    int ha = m_pTargetOffset->GetValue();
+    int dec = m_pTargetDec->GetValue();
+    bool sideEast = m_pTargetEast->GetValue() == 1;
+    if (sideEast)
+        ha = -ha;
+    CalCustomDialog custDlg(this, ha, dec);
+    custDlg.ShowModal();            // Dialog handles the UI updates
+}
+
+CalCustomDialog::CalCustomDialog(CalibrationAssistant* Parent, int DefaultHA, int DefaultDec)
+    : wxDialog(pFrame, wxID_ANY, _("Save Customized Calibration Position"),
+    wxDefaultPosition, wxSize(474, -1), wxCAPTION | wxCLOSE_BOX)
+{
+    m_Parent = Parent;
+    wxStaticBoxSizer* tgtSizer = new wxStaticBoxSizer(wxVERTICAL, this, _("Target Position"));
+    wxFlexGridSizer* targetPosSizer = new wxFlexGridSizer(1, 5, 5, 15);
+
+    int spinnerWidth = StringWidth(this, "0000");
+    m_pTargetDec = NewSpinnerInt(this, wxSize(spinnerWidth, -1), DefaultDec, -50, 50, 5, _("Target declination for slew, as close to Dec = 0 as possible for your location"));
+    AddTableEntryPair(this, targetPosSizer, _("Declination"), m_pTargetDec);
+    m_pTargetOffset = NewSpinnerInt(this, wxSize(spinnerWidth, -1), abs(DefaultHA), 5, 50, 5, _("Target offset from central meridian, in degrees; east or west based on 'Pointing' buttons"));
+    AddTableEntryPair(this, targetPosSizer, _("Meridian offset (degrees)"), m_pTargetOffset);
+
+    wxStaticBoxSizer* sizerTargetSOP = new wxStaticBoxSizer(wxHORIZONTAL, this, _("Pointing"));
+    m_pTargetWest = new wxRadioButton(this, wxID_ANY, _("West"), wxDefaultPosition, wxDefaultSize, wxRB_GROUP);
+    m_pTargetWest->SetToolTip(_("Scope on the east side of pier, pointing west"));
+    m_pTargetWest->Bind(wxEVT_COMMAND_RADIOBUTTON_SELECTED, &CalCustomDialog::OnTargetWest, this);
+    m_pTargetEast = new wxRadioButton(this, wxID_ANY, _("East"));
+    m_pTargetEast->SetToolTip(_("Scope on the west side of pier, pointing east"));
+    m_pTargetEast->Bind(wxEVT_COMMAND_RADIOBUTTON_SELECTED, &CalCustomDialog::OnTargetEast, this);
+    if (DefaultHA <= 0)
+        m_pTargetEast->SetValue(true);
+    else
+        m_pTargetWest->SetValue(true);
+    sizerTargetSOP->Add(m_pTargetWest);
+    sizerTargetSOP->Add(m_pTargetEast);
+    targetPosSizer->Add(sizerTargetSOP);
+    tgtSizer->Add(targetPosSizer);
+
+    m_pEastWestOnly = new wxCheckBox(this, wxID_ANY, wxEmptyString);
+    m_pEastWestOnly->SetToolTip(_("Check this if calibration can only be done on a particular side of the meridian because of nearby obstructions"));
+    if (m_pTargetWest->GetValue())
+        m_pEastWestOnly->SetLabelText(_("Western sky only"));
+    else
+        m_pEastWestOnly->SetLabelText(_("Eastern sky only"));
+
+    wxStaticText* pMessage = new wxStaticText(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize(600, -1), wxALIGN_CENTER_HORIZONTAL);
+    pMessage->SetLabelText(_("If your site location requires a unique sky position for calibration, ") +
+        _("you can specify it here."));

```

Comment:
This needs to be a single translatable string, not two separate strings -- the individual pieces may not make any sense translated then concatenated together.
The translator is going to use poedit and get a list of all the strings marked by `_()` and translate them one by one.
The two fragments make sense in English concatenated together, but that does not necessarily mean it will work in other languages. For example, It may be more idiomatic in some other language to phrase it like "specify here the sky position required by your location conditional" but as you have it here it forces the translator to translate the individual pieces ("if your site...", "you can spcify") then concatenates them together which prevents the translator from coming up with the best wording for the full sentence.
```suggestion
    pMessage->SetLabelText(_("If your site location requires a unique sky position for calibration, "
                             "you can specify it here."));
```